cli/src/program.rs

```diff
@@ -1478,6 +1515,72 @@ fn process_dump(
     }
 }
 
+fn process_dump_owned_accounts(
+    rpc_client: &RpcClient,
+    config: &CliConfig,
+    account_pubkey: Option<Pubkey>,
+    output_directory: &str,
+) -> ProcessResult {
+    if let Some(account_pubkey) = account_pubkey {
+        if let Some(account) = rpc_client
+            .get_account_with_commitment(&account_pubkey, config.commitment)?
+            .value
+        {
+            if account.owner == bpf_loader::id()
+                || account.owner == bpf_loader_deprecated::id()
+                || account.owner == bpf_loader_upgradeable::id()
+            {
+                let program_accounts = rpc_client.get_program_accounts(&account_pubkey);
```

Comment:
How about doing a `fs::create_dir_all` to the `output_directory`? That seems nicer than expecting users to create the directory themselves, unless `File::create` does that automatically